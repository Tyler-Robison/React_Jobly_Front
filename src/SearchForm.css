#searchButton {
    background-color: rgb(31, 150, 31);
    font-weight: 700;
}