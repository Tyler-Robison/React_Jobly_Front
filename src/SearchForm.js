import { useState } from "react"
import JoblyApi from "./API";
import { Button, Input } from "@material-ui/core";
import SearchIcon from '@mui/icons-material/Search';
import './SearchForm.css'


const SearchForm = ({ setCompanies, setJobs }) => {
    const INITIAL_STATE = {
        text: ''
    }
    const [formData, setFormData] = useState(INITIAL_STATE);

    const filter = async (filterTerm) => {
        if (setCompanies) {
            const res = await JoblyApi.getCompanies(filterTerm)
            setCompanies(res)
        } else {
            const res = await JoblyApi.getJobs(filterTerm)
            setJobs(res)
        }
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        filter(formData.text)
        setFormData(INITIAL_STATE)
    }

    const handleChange = (e) => {
        const { name, value } = e.target
        setFormData(formData => ({
            ...formData, [name]: value
        }))
    }

    const resetList = async () => {
        if (setCompanies) {
            const res = await JoblyApi.getCompanies()
            setCompanies(res)
        } else {
            const res = await JoblyApi.getJobs()
            setJobs(res)
        }
    }

    return (
        <>
            <form onSubmit={handleSubmit}>
                <Input
                    // color='secondary'
                    variant='Outlined'
                    type='text'
                    name='text'
                    value={formData.text}
                    onChange={handleChange}
                />

                <Button
                    type="submit"
                    variant="contained"
                    id='searchButton'
                    endIcon={<SearchIcon />}
                >Search</Button>
            </form>

            <Button onClick={resetList}
                variant="contained"
                color="primary"
            >Undo Search</Button>
        </>
    )
}

export default SearchForm;