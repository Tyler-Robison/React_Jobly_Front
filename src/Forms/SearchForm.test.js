import { render, screen } from '@testing-library/react';
import SearchForm from './SearchForm';
import { MemoryRouter } from "react-router-dom";
import UserProvider from '../testUtils'


it("renders without crashing", function () {
  <MemoryRouter>
    <SearchForm />
  </MemoryRouter>
});

it("matches snapshot", function () {

    const container = render(
        <MemoryRouter>
            <UserProvider>
                <SearchForm />
            </UserProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});