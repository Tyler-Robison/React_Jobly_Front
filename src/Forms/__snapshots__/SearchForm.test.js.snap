// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline"
      variant="Outlined"
    >
      <input
        class="MuiInputBase-input MuiInput-input"
        name="text"
        type="text"
        value=""
      />
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      id="searchButton"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Search
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Undo Search
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
